cmd/gb-vendor/vendor/repo.go

```diff
@@ -154,7 +153,17 @@ func (g *gitrepo) URL() string {
 	return g.url
 }
 
-func (g *gitrepo) Checkout(branch, revision string) (WorkingCopy, error) {
+// Checkout fetchs the remote branch, and optionaly updates to the supplied
+// tag or revision. If both are supplied, an error is returned. If the branch
+// is blank, then the default remote branch will be used. If the branch is
+// "HEAD", an error will be returned.
+func (g *gitrepo) Checkout(branch, tag, revision string) (WorkingCopy, error) {
+	if branch == "HEAD" {
+		return nil, fmt.Errorf("cannot checkout HEAD")
```

Comment:
given this can only happen when `gb vendor update $SOMETHING_TAGGED` I could change this to

"cannot update dependency that has been previously fetched with -tag or -revision"